versions/1/integration/elasticsearch/spec.yml

```diff
@@ -28,3 +28,49 @@ spec:
           pattern: '^.+\.json$'
           contentMediaType: "application/json"
           required: false
+    - description: Folder containing Elasticsearch Transforms
+      # https://www.elastic.co/guide/en/elasticsearch/reference/current/transforms.html
+      type: folder
+      name: transform
+      required: false
+      contents:
+        - description: |-
+            An Elasticsearch transform folder.
+            Upon installation of a package that contains a transform asset, the create behavior is as follows:
+            1. If a destination index template is included:
+              1. Create the destination index template.
+              2. Add the appropriate _meta fields to mark the matching indices as managed by Fleet.
+              3. If a fields directory is included:
+                1. Use the defined fields as the destination index template mappings.
+              4. if the destination index doesn't already exist:
+                1. Initialize the index.
+            3. Create the transform in the default namespace using kibana_system privileges.
+            5. Start the transform
+
+            Upon upgrading a package that contains a transform asset, the upgrade behavior is as follows:
+            1. If a destination index template is included:
+              1. Delete the destination index template.
+              2. Matching indices are not deleted.
+            2. Stop the transform.
+            3. Delete the transform.
+            4. From here, follow the creation behavior.
+          type: folder
+          pattern: '^[a-z0-9_]+$'
+          contents:
+            - description: Elasticsearch transform file
+              type: file
+              required: true
+              contentMediaType: "application/x-yaml"
+              pattern: '^transform\.yml$'
```

Comment:
Should we have validation on the content of this file? For instance, I'd like there to be constraints on the source / destination index names to match the data stream naming scheme.

Reply:
I am worried that validation on source/dest index names will make this less applicable to use of transforms in other packages. Security team are not using datastream naming scheme in their host risk score package. Can we avoid strict naming scheme checking?

Reply:
I've added validations on the transform based on the [schema from Elasticsearch's repo](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/transform/src/test/resources/rest-api-spec/schema/transform_config.schema.json).

Currently no constraints on the source / destination index names as following @sophiec20's comment I thought we might want and wait with that for the next iteration.

The PR includes the infrastructure for including such constraints on the next iteration:
[destination](https://github.com/elastic/package-spec/pull/307#discussion_r906832249), [source](https://github.com/elastic/package-spec/pull/307#discussion_r906832321).

Reply:
Discussed with @sophiec20 on Slack. It seems we have some migration scenarios of existing transforms to packages, and this restriction won't be feasible for such cases. Let's move this to a recommendation in the spec's description and remove the TODOs for now. Thanks @eyalkraft

Reply:
Done